client/src/app/site/motions/services/motion-poll-pdf.service.ts

```diff
@@ -0,0 +1,283 @@
+import { MotionRepositoryService } from './motion-repository.service';
+import { PdfDocumentService } from 'app/core/services/pdf-document.service';
+import { UserRepositoryService } from 'app/site/users/services/user-repository.service';
+
+/**
+ * Creates a pdf for a motion poll. Takes as input any motionPoll
+ * Provides the public method `printBallots(motionPoll)` which should be convenient to use.
+ *
+ * @example
+ * ```ts
+ * this.MotionPollPdfService.printBallos(this.poll);
+ * ```
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class MotionPollPdfService {
+    /**
+     * The method to determine the number of ballots to print. Value is
+     * decided in configuration service as `motions_pdf_ballot_papers_selection`.
+     * Options are:
+     *
+     * - NUMBER_OF_DELEGATES Amount of users belonging to the predefined 'delegates' group (group id 2)
+     * - NUMBER_OF_ALL_PARTICIPANTS The amount of all registered users
+     * - CUSTOM_NUMBER a given number of ballots (see {@link ballotCustomCount})
+     */
+    private ballotCountSelection: 'NUMBER_OF_DELEGATES' | 'NUMBER_OF_ALL_PARTICIPANTS' | 'CUSTOM_NUMBER';
+
+    /**
+     * An arbitrary number of ballots to print, if {@link ballotCountSection} is set
+     * to CUSTOM_NUMBER. Value is fetched from the configuration value `motions_pdf_ballot_papers_number`
+     */
+    private ballotCustomCount: number;
+
+    /**
+     * The event name (as set in config `general_event_name`)
+     */
+    private eventName: string;
+
+    /**
+     * The base64 logo to be printed (if set in config `logo_pdf_ballot_paper`)
+     * Will be loaded asynchronously and inserted into pdfMake as `ballot-logo.any`
+     */
+    private b64Logo: string;
+
+    /**
+     * Constructor. Subscribes to configuration values
+     *
+     * @param translate handle translations
+     * @param motionRepo get parent motions
+     * @param configService Read config variables
+     * @param userRepo User repository for counting amount of ballots needed
+     * @param pdfService the pdf document creation service
+     */
+    public constructor(
+        private translate: TranslateService,
+        private motionRepo: MotionRepositoryService,
+        private configService: ConfigService,
+        private userRepo: UserRepositoryService,
+        private pdfService: PdfDocumentService
+    ) {
+        this.configService.get('motions_pdf_ballot_papers_number').subscribe(count => (this.ballotCustomCount = count));
+        this.configService
+            .get('motions_pdf_ballot_papers_selection')
+            .subscribe(selection => (this.ballotCountSelection = selection));
+        this.configService.get('general_event_name').subscribe(name => (this.eventName = name));
+        this.configService.get('logo_pdf_ballot_paper').subscribe(url => {
+            if (url.path) {
+                this.pdfService.convertUrlToBase64(url.path).then(b64image => (this.b64Logo = b64image));
```

Comment:
this might be a problem, since the service will "permanently" hold the (very long) base64 in it's cache.

We might change this to only temporarily create and hold the base64 after the user clicks on "create pdf" instead of keeping it saved in the service